packages/api-client/src/components/ImportCollection/PrefetchError.vue

```diff
@@ -1,40 +1,53 @@
 <script setup lang="ts">
 import type { PrefetchResult } from '@/components/ImportCollection/hooks/useUrlPrefetcher'
-import { ScalarIcon, useModal } from '@scalar/components'
-import { ref } from 'vue'
+import { isUrl } from '@/libs'
+import { ScalarIcon } from '@scalar/components'
+import { computed } from 'vue'
 
-defineProps<{
-  prefetchResult: PrefetchResult
+const props = defineProps<{
+  result?: PrefetchResult
 }>()
 
-const errorModalState = useModal()
+const domain = computed(() => {
+  if (!props.result?.input || !isUrl(props.result.input)) {
+    return undefined
+  }
 
-const errorMessage = ref('')
+  const hostname = new URL(props.result?.input)?.hostname
 
-const handleExpandError = (message: string) => {
-  errorMessage.value = message
-  errorModalState.show()
-}
+  const CHARACTER_LIMIT = 24
+
+  return hostname?.length > CHARACTER_LIMIT
+    ? `${hostname.slice(0, CHARACTER_LIMIT)}...`
+    : hostname
+})
 </script>
 
 <template>
-  <template v-if="prefetchResult.error">
+  <template v-if="result?.error">
+    <div
+      v-if="result?.input && domain"
+      class="text-sm break-words text-center w-full font-medium">
+      We couldn't find an OpenAPI document at this URL. You can try visiting
+      <a
+        class="max-w-[50ch] truncate"
+        :href="result.input"
+        rel="noopener nofollow"
+        target="_blank"
+        v-text="domain" />
+      to download the OpenAPI file manually.
+    </div>
```

Comment:
what do you think about leading to the action?

```suggestion
      We couldn’t find an OpenAPI document at the provided URL. Please import the
      <a
        class="max-w-[50ch] truncate"
        :href="result.input"
        rel="noopener nofollow"
        target="_blank"
        v-text="domain" />
      OpenAPI document manually.
    </div>
```

and then add the `Import Collection` that would open the import modal?

Reply:
<img width="506" alt="Screenshot 2024-10-30 at 14 28 41" src="https://github.com/user-attachments/assets/be8a84d6-c9cc-4a7a-be68-89f33d027a87">